vsock-proxy/enclave/src/app_configuration.rs

```diff
@@ -44,29 +43,25 @@ const CREDENTIALS_FILE: &str = "credentials.bin";
 
 const LOCATION_FILE: &str = "location.txt";
 
+const APPLICATION_CONFIGURATION_ID_LENGTH: usize = 32;
+
 pub(crate) fn setup_application_configuration<T>(
     em_app_credentials: &EmAppCredentials,
     ccm_backend_url: &CCMBackendUrl,
     api: T,
     fs_root: &Path,
-    app_config_id: Blob
+    app_config_id: &str
 ) -> Result<(), String>
 where
     T: ApplicationConfiguration,
 {
     info!("Requesting application configuration.");
 
+    let app_config_id_raw = <[u8; APPLICATION_CONFIGURATION_ID_LENGTH]>::try_from(app_config_id.as_bytes()).map_err(|err| format!("Cannot convert application config id into an array. Error: {:?}. Perhaps your id isn't {} characters long?", err, APPLICATION_CONFIGURATION_ID_LENGTH))?;
```

Comment:
I understand your confusion and I initially had this field as a byte array, but then I have found another function in `em-app` that is used in Salmiac and it requires a `&str`. The function in question is: https://github.com/fortanix/rust-sgx/blob/d446f203a6929d85250d5f21fe0e4bcf2f91efb9/em-app/src/lib.rs#L110

Reply:
The code change you propose doesn't work because `try_from` is an associated function.
```
error[E0599]: no method named `try_from` found for reference `&[u8]` in the current scope
   --> enclave/src/app_configuration.rs:61:10
    |
60  |       let app_config_id_raw: [u8; APPLICATION_CONFIGURATION_ID_LENGTH] = app_config_id.as_bytes()
    |  ________________________________________________________________________-
61  | |         .try_from()
    | |         -^^^^^^^^ this is an associated function, not a method
    | |_________|
    |
    |
    = note: found the following associated functions; to be used as methods, functions must have a `self` parameter
note: the candidate is defined in the trait `TryFrom`
   --> /home/nikitashyrei/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/convert/mod.rs:687:5
    |
687 |     fn try_from(value: T) -> Result<Self, Self::Error>;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: use associated function syntax instead
    |
60  |     let app_config_id_raw: [u8; APPLICATION_CONFIGURATION_ID_LENGTH] = [u8]::try_from()
    |                                                                        ~~~~~~~~~~~~~~~~
help: there is a method `from` with a similar name
    |
61  |         .from()
    |          ~~~~


```